packages/ui-library/src/components/select-field/index.js

```diff
@@ -9,7 +9,7 @@ import { forwardRef } from "@wordpress/element";
  * @param {JSX.Element} error Error node.
  * @param {string} [className] Optional CSS class.
  * @param {boolean} [disabled] Disabled state.
- * @param {string} label Label.
+ * @param {string | JSX.Element} label Label.
```

Comment:
The `label` is made to be _only_ a string for a11y reasons.
I think we should reconsider this changing it back, at the very least discuss it with our a11y expert 😉 

Checking the history on GitHub I see there unfortunately not much "why" added to the PR. Only a11y reasons: https://github.com/Yoast/wordpress-seo/pull/18330
I suppose I still don't know and suspect screen readers read HTML out loud. Making for a confusing experience.